src/data/graphics/trainers.h

```diff
@@ -292,116 +292,110 @@ const u8 gTrainerBackPic_Steven[] = INCBIN_U8("graphics/trainers/back_pics/steve
 const u32 gTrainerBackPicPalette_Red[] = INCBIN_U32("graphics/trainers/back_pics/red.gbapal.lz");
 const u32 gTrainerBackPicPalette_Leaf[] = INCBIN_U32("graphics/trainers/back_pics/leaf.gbapal.lz");
 
-static const union AnimCmd *const sAnims_Trainer[] ={
-    sAnim_GeneralFrame0,
-};
-
-#define TRAINER_SPRITE(trainerPic, file, x, y, rotation)                                  \
+#define TRAINER_SPRITE(trainerPic, file, ...)                                             \
     [TRAINER_PIC_##trainerPic] =                                                          \
     {                                                                                     \
-        .y_offset = 8,                                                                    \
         .frontPic = {gTrainerFrontPic_##file, TRAINER_PIC_SIZE, TRAINER_PIC_##trainerPic},\
         .palette = {gTrainerPalette_##file, TRAINER_PIC_##trainerPic},                    \
-        .animation = sAnims_Trainer,                                                      \
-        .mugshotCoords = {x, y},                                                          \
-        .mugshotRotation = rotation,                                                      \
+        .mugshotCoords = {DEFAULT(0, __VA_ARGS__), DEFAULT_2(0, __VA_ARGS__)},            \
+        .mugshotRotation = DEFAULT_3(0x200, __VA_ARGS__),                                 \
     }
 
 const struct TrainerSprite gTrainerSprites[] =
 {
-    TRAINER_SPRITE(HIKER, Hiker, 0, 0, 0x200),
```

Comment:
I do wonder if this is not a small to far with the macro. I see that it's redundant information but how important would it be for the user to know? Would the example with the E4 members be enough to get the gist?

Reply:
Mayhaps a comment could be addede a giving brief explanation of the purpose and contents of the array.

Like for example:

```diff
// This array controls the basic parameters of a trainer sprite. It uses a
// macro called TRAINER_SPRITE to simplify the depiction of data.
// The first 2 parameters invoke a trainer's front pic and palette by
// calling a "TRAINER_PIC" constant (Ex: TRAINER_PIC_HIKER), and
// a gTrainerFrontPic and gTrainerPalette pointers, like for example
// "gTrainerFrontPic_Hiker" and "gTrainerPalette_Hiker".
// The last 3 parameters default to "0, 0, 0x200"  which is a default
// value used by the majority of the game's trainer sprites.
// They control the X and Y coordinates of the mugshot on the
// screen, as well as their rotation.
const struct TrainerSprite gTrainerSprites[] =
```

Or something that is better written.

Reply:
something that is much more condensed but yeah

Reply:
> Or something that is better written.
Not sure if I made it better at all but there is now a description